overlord/snapstate/autorefresh_gating.go

```diff
@@ -213,6 +217,10 @@ func HoldRefresh(st *state.State, gatingSnap string, holdDuration time.Duration,
 			gating[heldSnap] = make(map[string]*holdState)
 		}
 		gating[heldSnap][gatingSnap] = hold
+
+		if durationMin == 0 || left < durationMin {
```

Comment:
do we have a unit test covering the `left < durationMin` case where one snap can be held for longer than another and we effectively return the sooner holding time? a bit hard to tell reading the unit test cases

Reply:
Yes, it's tested: in the first step of the test the snap is only holding itself which yields the maximum holding time, in next steps it holds itself and other snaps which reduced it to 48h and less.